python/src/com/jetbrains/python/debugger/PyDebugRunner.java

```diff
@@ -313,6 +302,23 @@ private void fillDebugParameters(@NotNull Project project,
     }
   }
 
+  public static void configureDebugEnvironment(@NotNull Project project, Map<String, String> environment) {
+    if (PyDebuggerOptionsProvider.getInstance(project).isSupportGeventDebugging()) {
+      environment.put(GEVENT_SUPPORT, "True");
+    }
+
+    PyDebuggerSettings debuggerSettings = PyDebuggerSettings.getInstance();
```

Comment:
They should because we set the environment variables before the debugger object is created.